.circleci/delete.py

```diff
@@ -0,0 +1,244 @@
+"""
+This script deletes your Review apps and their linked databases. It is intended
+to be run when PR's are merged and on a schedule.
+"""
+
+import sys
+from datetime import datetime, timedelta
+from time import sleep
+from gql import gql, Client
+from gql.transport.requests import RequestsHTTPTransport
+from settings import (
+    DASH_ENTERPRISE_HOST,
+    MAIN_BRANCHNAME,
+    REVIEW_BRANCHNAME,
+    REVIEW_APPNAME,
+    SERVICE_USERNAME,
+    SERVICE_API_KEY,
+    PREFIX,
+    PERIOD,
+    TIME,
+    LAST_UPDATE,
+)
+
+if sys.version_info[0:2] < (3, 6) or sys.version_info[0:2] > (3, 7):
+    raise Exception(
+        "This script has only been tested on Python 3.6. "
+        + "You are using {major}.{minor}.".format(
+            major=sys.version_info[0], minor=sys.version_info[1]
+        )
+    )
+
+transport = RequestsHTTPTransport(
+    url="https://{DASH_ENTERPRISE_HOST}/Manager/graphql".format(
+        DASH_ENTERPRISE_HOST=DASH_ENTERPRISE_HOST
+    ),
+    auth=(SERVICE_USERNAME, SERVICE_API_KEY),
+    use_json=True,
+    retries=5,
+)
+
+client = Client(transport=transport)
+
+
+PAGE = 0
+
+apps = []
+apps_result = []
+services = []
+
+if MAIN_BRANCHNAME != REVIEW_BRANCHNAME:
+    print("Querying apps...")
+    VAR_TYPE = "$page: Int"
+    VAR = "page: $page, allApps:true"
+    PARAMS = {"page": PAGE}
+else:
+    print("Querying review app...")
+    VAR_TYPE = "$name: String"
+    VAR = "name: $name"
+    PARAMS = {"name": REVIEW_APPNAME}
```

Comment:
I don't understand this logic. Could you add some comments explaining what we're doing and why? 

Does the `delete.py` schedule get run on every branch? Meaning if there are 5 branches then there are 5 schedules?

If so, then I don't think that our CI config is set up correctly. I think it should only run the schedule once on the `main` branch.